core/src/main/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java

```diff
@@ -98,24 +98,25 @@ public boolean followedLimitAdvice() {
     @Override
     public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
         long rowsRemaining;
+        int baseOrder = base.hasDescendingOrder() ? ORDER_DESC : ORDER_ASC;
         final int order;
         if (limitLoFunction != null) {
             limitLoFunction.init(frameSequence.getSymbolTableSource(), executionContext);
             rowsRemaining = limitLoFunction.getLong(null);
             // on negative limit we will be looking for positive number of rows
             // while scanning table from the highest timestamp to the lowest
             if (rowsRemaining > -1) {
-                order = ORDER_ASC;
+                order = baseOrder;
             } else {
-                order = ORDER_DESC;
+                order = reverse(baseOrder);
                 rowsRemaining = -rowsRemaining;
             }
         } else {
             rowsRemaining = Long.MAX_VALUE;
-            order = ORDER_ANY;
+            order = baseOrder;
```

Comment:
It used to be ANY when no limit specified, now it's ASC.
Why ANY is incorrect?

Reply:
Passing ANY to data frame factory 'below' made it ASC and caused the issue . 
It's either ASC or DESC now, and former should cover 'previous' ANY .   
I'm not a fan of deciding directions at this stage really.